### PR TITLE
MyPy errors for constructor args fixed

### PR DESCRIPTION
**Pull request by kumare3**

### CODE REVIEW

File: flytekit/annotated/interface.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def __init__(
                     self._inputs[k] = v
                 else:
                     self._inputs[k] = (v, None)
-        self._outputs = outputs
+        self._outputs = outputs if outputs else {}
```
Comment (wild-endeavor) [inline flytekit/annotated/interface.py:38]: this won't break anything right?

File: flytekit/taskplugins/sagemaker/hpo.py [hunk 2]
```diff
@@ -62,21 +61,20 @@ def __init__(
 
         self._task_config = task_config
         self._training_task = training_task
-        iface = training_task.python_interface
```
Comment (wild-endeavor) [inline flytekit/taskplugins/sagemaker/hpo.py:65]: why the move?  did mypy pick up on this?

**Review by wild-endeavor — APPROVED** [review]
approving with a couple questions.

File: flytekit/annotated/base_sql_task.py [hunk 3]
```diff
@@ -21,9 +21,8 @@ def __init__(
         task_type="sql_task",
         inputs: Optional[Dict[str, Type]] = None,
         metadata: Optional[TaskMetadata] = None,
-        task_config: T = None,
+        task_config: Optional[T] = None,
         outputs: Dict[str, Type] = None,
```
Comment (katrogan) [inline flytekit/annotated/base_sql_task.py:25]: this doesn't need to be optional too?

**Review by katrogan — APPROVED** [review]

File: flytekit/annotated/interface.py (hunk 1 above)
Comment (kumare3) [reply flytekit/annotated/interface.py:38]: nope

File: flytekit/taskplugins/sagemaker/hpo.py (hunk 2 above)
Comment (kumare3) [reply flytekit/taskplugins/sagemaker/hpo.py:65]: no, cleaner to have it next to where we use it

File: flytekit/annotated/base_sql_task.py (hunk 3 above)
Comment (kumare3) [reply flytekit/annotated/base_sql_task.py:25]: mypy complains :(

**Pull request merged**